dts/bindings/test/vnd,i3c.yaml

```diff
@@ -15,3 +15,9 @@ include:
 properties:
   label:
     type: string
+
+  supports-setaasa:
+    type: boolean
+    description: |
+      Indicates if the device supports the CCC SETAASA. If true, it will
+      be used as an optimization for bus initialization.
```

Comment:
@erwango 
The reason for adding this change was an issue I encountered during compilation when defining I3C targets in the overlay file for the samples. The error in the DTS reported something like :
' error: 'DT_N_S_soc_S_i3c_40005c00_S_i3cdev0_000000020813811000_P_supports_setaasa' undeclared here...'

Upon investigating the code, I found that the issue comming from the following macro in Zephyr:

```c #define I3C_DEVICE_DESC_DT(node_id)                \
    {                                               \
        .bus = DEVICE_DT_GET(DT_BUS(node_id)),      \
        .dev = DEVICE_DT_GET(node_id),              \
        .static_addr = DT_PROP_BY_IDX(node_id, reg, 0), \
        .pid = ((uint64_t)DT_PROP_BY_IDX(node_id, reg, 1) << 32) \
            | DT_PROP_BY_IDX(node_id, reg, 2),     \
        .init_dynamic_addr =                        \
            DT_PROP_OR(node_id, assigned_address, 0), \
        .supports_setaasa = DT_PROP(node_id, supports_setaasa), \
    }, ```


The problem happens because DT_PROP is used to directly fetch the supports_setaasa property, but if the property is missing, the macro does not handle it.

To fix the issue, we have two possible approaches:

1) Add the supports_setaasa property in the relevant yaml, as I did.
2) Propose a change in Zephyr to use DT_PROP_OR instead of DT_PROP. This way, the macro can provide a default value (e.g., 0) when the property is not explicitly defined in the device tree. 

which approach do you prefer ?

Reply:
Ok, hence the need for documenting commit messages ;).
I'd opt for option 3: Add this property in `i3c-controller.yaml`

Reply:
Yes, I can add that in a separate commit.
Just to clarify, the issue was raised for the i3c targets, not the controllers. The previous macro will be called when looping over the targets.
Since I used compatible: "vnd,i3c" for the targets, I have added the property in the vnd,i3c.yaml file